extensions/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentPuller.java

```diff
@@ -273,6 +276,65 @@ public boolean apply(Throwable e)
   }
 
   /**
+   * Gets the latest version using the "parent" of uri as a prefix. The "parent" of uri is evaluated assuming '/' delimited paths
+   *
+   * @param uri     The URI of interest whose "parent" will be searched as a key prefix for the latest version
+   * @param pattern The pattern matcher to determine if a *key* is of interest. This will match against the entire key,
+   *                not just the equivalent "filename" like some other implementations. A null value matches everything
+   *
+   * @return A URI to the most recently modified object which matched the pattern.
+   */
+  @Override
+  public URI getLatestVersion(final URI uri, final Pattern pattern)
+  {
+    try {
+      return RetryUtils.retry(
+          new Callable<URI>()
+          {
+            @Override
+            public URI call() throws Exception
+            {
+              final AmazonS3URI s3URI = new AmazonS3URI(checkURI(uri));
```

Comment:
As far as I know the S3 deep-storage only relies on on Jets3t classes, in order to stay compatible with non-AWS S3 compatible implementation. We only rely on AWS SDK to support AWS-specific things such as auto-scaling or authentication scheme. Is there a particular reason we need to rely on aws classes here?